WooCommerce/Classes/ViewRelated/Products/Variations/Add Attributes/AddAttributeViewController.swift

```diff
@@ -0,0 +1,298 @@
+import UIKit
+import Yosemite
+import WordPressUI
+
+final class AddAttributeViewController: UIViewController {
+
+    @IBOutlet private weak var tableView: UITableView!
+    private let ghostTableView = UITableView()
+
+    private let product: Product
+    private let viewModel: AddAttributeViewModel
+
+    /// Keyboard management
+    ///
+    private lazy var keyboardFrameObserver: KeyboardFrameObserver = KeyboardFrameObserver { [weak self] keyboardFrame in
+        self?.handleKeyboardFrameUpdate(keyboardFrame: keyboardFrame)
+    }
+
+    /// Init
+    ///
+    init(product: Product) {
```

Comment:
Would it make sense to pass the view model directly? That way this view controller wouldn't have to instantiate its own view model, which seems a bit off.

Anyway, it doesn't look like the view controller uses the product directly, so maybe it wouldn't need to retain it?

Reply:
Thank you @ctarda.

You are right, it's better that the viewmodel is instantiated outside. I followed your suggestions and made the changes here df5ccf25a76277e0d2ac3a831b0953f3b96d4327